src/resource/solidDataset.test.ts

```diff
@@ -2017,6 +2017,146 @@ describe("createContainerAt", () => {
     });
   });
 
+  describe("Creating non-empty container", () => {
+    function getMockUpdatedDataset(
+      changeLog: WithChangeLog["internal_changeLog"],
+      fromUrl: IriString
+    ): SolidDataset & WithChangeLog & WithResourceInfo {
+      const mockThing = addUrl(
+        createThing({ url: "https://arbitrary.vocab/subject" }),
+        "https://arbitrary.vocab/predicate",
+        "https://arbitrary.vocab/object"
+      );
+      let mockDataset = setThing(createSolidDataset(), mockThing);
+
+      changeLog.additions.forEach((tripleToAdd) => {
+        let additionThing =
+          getThing(mockDataset, tripleToAdd.subject.value) ??
+          createThing({ url: tripleToAdd.subject.value });
+        additionThing = addUrl(
+          additionThing,
+          tripleToAdd.predicate.value,
+          tripleToAdd.object.value
+        );
+        mockDataset = setThing(mockDataset, additionThing);
+      });
+
+      const resourceInfo: WithResourceInfo["internal_resourceInfo"] = {
+        sourceIri: fromUrl,
+        isRawData: false,
+      };
+
+      return {
+        ...mockDataset,
+        internal_changeLog: changeLog,
+        internal_resourceInfo: resourceInfo,
+      };
+    }
+
+    it("throws if container SolidDataset has updates", async () => {
+      const mockFetch = jest
+        .fn(window.fetch)
+        .mockReturnValue(
+          Promise.resolve(
+            mockResponse(undefined, { url: "https://arbitrary.pod/container/" })
+          )
+        );
+
+      const mockDataset = getMockUpdatedDataset(
+        {
+          additions: [
+            DataFactory.quad(
+              DataFactory.namedNode("https://some.vocab/subject"),
+              DataFactory.namedNode("https://some.vocab/predicate"),
+              DataFactory.namedNode("https://some.vocab/object"),
+              undefined
+            ),
+          ],
+          deletions: [],
+        },
+        "https://arbitrary.pod/container/"
+      );
+
+      await expect(
+        createContainerAt(
+          "https://arbitrary.pod/container/",
+          {
+            fetch: mockFetch,
+          },
+          mockDataset
```

Comment:
```suggestion
          {
            fetch: mockFetch,
            initialContent: mockDataset
          }
```
See the suggested signature change below.